src/core/server/server.ts

```diff
@@ -153,6 +153,10 @@ export class Server {
       metrics: metricsSetup,
     };
 
+    if (this.env.cliArgs.dryRunMigration) {
+      return coreSetup;
+    }
```

Comment:
This is exactly the kind of hijacking mechanism I would like to avoid by having a specific class/script for the dry run migration instead of bootstrapping the whole server. OTOH this means duplicating quite a big part of the `Server` class and associated bootstrap scripts, so I'm not sure which way is the best regarding maintainability. What do the others think?

Reply:
Dry run migrations rely on the fact that we're reproducing the behaviour of a normal migration 100% including how we load config, discover plugins, setup elasticsearch etc. While it's easy to write an integration test that asserts that a dry run does everything except start plugins, it's harder to write a test that a e.g. a `DryRunServer` class mimics the exact behaviour of the `Server`.